app/controllers/facility_reservations_controller.rb

```diff
@@ -80,6 +80,8 @@ def update
       end
     end
 
+    @reservation.assign_category_attribute(session_user, params[:reservation])
```

Comment:
Also, when you're in this context, I'm 95% the user is already authorized, so I don't think we need the additional check.